framework/db/BaseActiveRecord.php

```diff
@@ -1304,10 +1304,14 @@ public function unlink($name, $model, $delete = false)
             $p1 = $model->isPrimaryKey(array_keys($relation->link));
             $p2 = static::isPrimaryKey(array_values($relation->link));
             if ($p2) {
-                foreach ($relation->link as $a => $b) {
-                    $model->$a = null;
+                if ($delete) {
+                    $model->delete();
```

Comment:
Yes, you are right, since `$relation` is not affected at this program flow it is correct.